server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -680,7 +684,8 @@ private IndexMetadata(
         final IndexVersion indexCompatibilityVersion,
         @Nullable final IndexMetadataStats stats,
         @Nullable final Double writeLoadForecast,
-        @Nullable Long shardSizeInBytesForecast
+        @Nullable Long shardSizeInBytesForecast,
+        final Map<String, List<String>> inferenceModelsForFields
```

Comment:
New field for IndexMetadata that has the correspondence of models and the list of fields that refer to that model. It is serialized both using StreamInput / Output and toXContent so it can be retrieved from disk